### PR TITLE
chore(native filters): Expandable filter config modal

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
The native filters modal width is unnecessarily narrow which results in a suboptimal UX—excessive scrolling—especially when either the filter names and/or datasets/column names are long.
This commit adds an option to make the  'Add and edit filters' modal wider.

Sales_Dashboard

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After:

https://github.com/apache/superset/assets/1392866/6027a46b-aa5f-4f1d-8656-52d7c33c9928

Before:

https://github.com/apache/superset/assets/1392866/7d6311ac-2fa8-4b9b-8656-1e0dd9b67f43

### TESTING INSTRUCTIONS
Go to dashboard and choose a dataset contains a long name column name

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by john-bodley** [comment]
@justinpark in addition to the scroll bars, is there merit in making the default modal size wider?

**Comment by justinpark** [comment]
> in addition to the scroll bars, is there merit in making the default modal size wider?

Sure. I can adjust the default width from 50% to 70%

File: superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigModal.tsx [hunk 1]
```diff
@@ -580,8 +601,26 @@ function FiltersConfigModal({
...
+          <div>{t('Add and edit filters')}</div>
+          <ToggleIcon
+            iconSize="l"
+            iconColor={theme.colors.grayscale.base}
+            onClick={toggleExpand}
+          />
+        </div>
+      }
+      width="70%"
+      {...(expanded && { width: '100%', height: '100%' })}
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigModal.tsx:622]: Given that `StyleModalWrapper` is already receiving the `expanded` property, could you deal with `width` and `height` there and remove these two lines?

**Comment by justinpark** [comment]
@michael-s-molina could you review the update?

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by kasiazjc** [comment]
Thanks for the change @justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap?
image

I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?

**Comment by justinpark** [comment]
> Thanks for the change justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap? image
> 
> I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?

Sounds good. I updated the location of the button to the next of the footer buttons.
Screenshot 2023-07-12 at 1 33 09 PM

I also updated the default width to 880px as well.

**Comment by kasiazjc** [comment]
> > Thanks for the change justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap? image
> > I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?
> 
> Sounds good. I updated the location of the button to the next of the footer buttons. Screenshot 2023-07-12 at 1 33 09 PM
> 
> I also updated the default width to 880px as well.

Amazing, thank you!

**Pull request merged**